### DISCUSSION TITLE
Exchange equivalent to switch with default

### DISCUSSION THREAD

**Discussion by dandraka** [rabbitmq/rabbitmq-server#8186] [Questions] [answered]
Hi all

I'm trying to route a message in a way similar to a switch statement. If there's a matching binding I need it routed there, of course, but if -and **_only_** if- there's not I'd have it routed to a "default" queue instead of it getting discarded.

That's a behaviour similar to a switch, like the following pseudocode:

```
switch (animal) {
        case "DOG":
            queue = "dog-queue"; 
            break;
        case "CAT":
            queue = "cat-queue"; 
            break;
        default:
            queue = "otheranimals-queue"; 
            break;
    }
```

From my understanding, the topic and header exchange types come close to this, but don't have any provision for the "default" (i.e. not matching anything else) case.

Is there any way or at least workaround to achieve that?

Many thanks and regards,

Jim

**Comment by mkuratczyk** [comment, marked as answer]
Alternate exchange can do this: https://www.rabbitmq.com/ae.html

**Reply by dandraka** [reply]
That's exactly what I was looking for. Thanks :-)
